src/v3/test/integration/enroll-profile-update.test.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import mockResponse from '@okta/mocks/data/idp/idx/enroll-profile-submit.json';
+import { waitFor } from '@testing-library/preact';
+import { setup } from './util';
+
+describe('enroll-profile-update', () => {
+  it('should render form', async () => {
+    const { container, findByText } = await setup({ mockResponse });
+    await findByText(/Sign in/);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should send correct payload', async () => {
+    const {
+      authClient, user, findByTestId, findByText,
+    } = await setup({ mockResponse });
+
+    await findByText(/Sign in/);
+
+    const submitButton = await findByText('Submit', { selector: 'button' });
```

Comment:
Can you try `findByRole('button', { name: 'Submit' });`? The order of the queries in the RTL docs is ordered from preferred to least preferred

Reply:
Same above, you might be able to do `findByRole('heading', { level: 2 })` for the title text.